File: app/src/main/java/com/michel/pointscredit/base/PCBaseActivity.java
package com.michel.pointscredit.base;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.view.Window;

import com.michel.pointscredit.utils.MatrialStatusBarUtils;
import com.qmuiteam.qmui.widget.dialog.QMUITipDialog;

import butterknife.ButterKnife;
import butterknife.Unbinder;

/**
 * @Annotation <p>描述</p>
 * @Auth Sunny
 * @date 2018/3/16
 * @Version V1.0.0
 */

public abstract class PCBaseActivity extends AppCompatActivity implements View.OnClickListener {

    public Context mContext;
    private Unbinder unbinder;
    public QMUITipDialog mLoading;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        mContext = this;
        //沉浸式
        setContentView(getLayoutId());
        MatrialStatusBarUtils.compat(this);
        unbinder = ButterKnife.bind(this);
        initParams();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        if (unbinder != null) {
            unbinder.unbind();
        }

        if (mLoading != null && mLoading.isShowing()){
            mLoading.dismiss();
            mLoading = null;
        }
    }

    public void showLoading(){
        if (mLoading == null){
            mLoading = new QMUITipDialog.Builder(this)
                    .setIconType(QMUITipDialog.Builder.ICON_TYPE_LOADING)
                    .create();
        }
        if (mLoading != null && !mLoading.isShowing()){
            mLoading.show();
        }
    }

    public void dismissLoading(){
        if (mLoading != null && mLoading.isShowing()){
            mLoading.dismiss();
            mLoading = null;
        }
    }

    public void initParams() {
        if (mLoading == null){
//            mLoading = new QMUITipDialog.Builder(this)
//                    .setIconType(QMUITipDialog.Builder.ICON_TYPE_LOADING)
//                    .create();
        }

    }

    protected abstract int getLayoutId();

    /**
     * 設置點擊監聽
     */
    public void setViewsOnClickListener(View... views) {
        if (views != null && views.length > 0) {
            for (View view : views) {
                view.setOnClickListener(this);
            }
        }
    }

    @Override
    public void onClick(View view) {

    }

}

File: app/src/main/java/com/michel/pointscredit/base/PCBaseActivity.kt
package com.michel.pointscredit.base

import android.content.Context
import android.graphics.Color
import android.os.Build
import android.os.Bundle
import android.support.v4.view.ViewPager
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.view.Window
import butterknife.ButterKnife
import butterknife.Unbinder
import android.view.WindowManager
import com.michel.pointscredit.R
import com.michel.pointscredit.utils.MatrialStatusBarUtils
import com.qmuiteam.qmui.util.QMUIStatusBarHelper


/**
 * Created by 80010651 on 2018/3/13.
 * Base界面：
 * 这里主要是提供一些模板操作
 */
abstract class PCBaseActivity : AppCompatActivity(), View.OnClickListener {

    var unbinder: Unbinder? = null
    var mContext: Context? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        requestWindowFeature(Window.FEATURE_NO_TITLE)
//        QMUIStatusBarHelper.translucent(this) // 沉浸式状态栏
        //当系统版本为4.4或者4.4以上时可以使用沉浸式状态栏
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
            //透明状态栏
            window.addFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS)
            //透明导航栏
            window.addFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_NAVIGATION)
        }

        mContext = this
        setContentView(getLayoutId())
        unbinder = ButterKnife.bind(this)
        initParams()
    }

    override fun onDestroy() {
        super.onDestroy()
    }

    /**
     * 初始化参数
     */
    fun initParams() {

    }

    abstract fun getLayoutId(): Int

    /**
     * 設置點擊監聽
     */
    fun setViewsOnClickListener(vararg views: View) {
        if (views.isNotEmpty()) {
            for (view in views) {
                view.setOnClickListener(this)
            }
        }
    }

    override fun onClick(p0: View?) {

    }
}